pints/toy/_german_credit.py

```diff
@@ -0,0 +1,124 @@
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import numpy as np
+from . import ToyLogPDF
+import io
+import urllib
+import urllib.request
+from scipy import stats
+
+
+class GermanCreditLogPDF(ToyLogPDF):
+    r"""
+    Toy distribution based on a logistic regression model, which takes the
+    form,
+
+    .. math::
+
+        f(x, y|\beta) \propto \text{exp}(-\sum_{i=1}^{N} \text{log}(1 +
+        \text{exp}(-y_i x_i.\beta)) - \beta.\beta/2\sigma^2)
+
+    The data :math:`(x, y)` are a matrix of individual predictors (with 1s in
+    the first column) and responses (1 if the individual should receive credit
+    and -1 if not) respectively; :math:`\beta` is a 25x1 vector of coefficients
+    and :math:`\sigma^2=100`. The dataset here is from [1]_ but the test
+    problem is defined in [2]_.
+
+    Extends :class:`pints.LogPDF`.
+
+    Parameters
+    ----------
+    beta : float
+        vector of coefficients of length 25.
+
+    References
+    ----------
+    .. [1] "UCI machine learning repository", 2010. A. Frank and A. Asuncion.
+           https://archive.ics.uci.edu/ml/datasets/statlog+(german+credit+data)
+    .. [2] "The No-U-Turn Sampler:  Adaptively Setting Path Lengths in
+           Hamiltonian Monte Carlo", 2014, M.D. Hoffman and A. Gelman.
+    """
+    def __init__(self, x=None, y=None, download=False):
+        if x is None or y is None:
+            if download is False:
+                raise ValueError("No data supplied. Consider setting " +
+                                 "download to True to download data.")
+            x, y = self.download_data()
+            dims = x.shape[1]
+        else:
+            if download is True:
+                raise ValueError("Either supply no data or set " +
+                                 "download to True to download data, but " +
+                                 "not both.")
+            dims = x.shape[1]
+            if dims != 25:
+                raise ValueError("x must have 25 predictor columns.")
+            if max(y) != 1 or min(y) != -1:
+                raise ValueError("Output must be either 1 or -1.")
+        self._x = x
+        self._y = y
+        self._n_parameters = dims
+        self._sigma = 10
+        self._sigma_sq = self._sigma**2
+        self._N = len(y)
+
+    def __call__(self, beta):
+        log_prob = sum(-np.log(1 + np.exp(-self._y * np.dot(self._x, beta))))
+        log_prob += -1 / (2 * self._sigma_sq) * np.dot(beta, beta)
+        return log_prob
+
+    def data(self):
+        """ Returns data used to fit model. """
+        return self._x, self._y
+
+    def download_data(self):
+        """ Downloads data from [1]. """
+        url="http://archive.ics.uci.edu/ml/machine-learning-databases/statlog/german/german.data-numeric" # noqa
```

Comment:
You can write e.g.
```
url = ('https://lalalala/123'
       '/more/stuff/here')
```

Reply:
Thanks -- been looking for a better way to handle these!